src/tools/generate/generate_chapters.js

```diff
@@ -22,14 +36,53 @@ const generate_chapters = async () => {
 
       const markdown = await fs.readFile(file, 'utf-8');
       const { metadata, body, toc } = await parse_file(markdown);
+      check_meta_dates(metadata);
       await write_template(language, year, chapter, metadata, body, toc);
+      await write_sitemap(language, year, chapter, metadata);
     } catch (error) {
       console.error(error);
       console.error('  Failed to generate chapter, moving onto the next one. ');
     }
   }
+  await write_sitemap_end();
 };
 
+const read_config = async (year) => {
+  try {
+  config_file = fs.readFileSync('config/'+year+'.json');
```

Comment:
I know it doesn't *really* matter in this context, but perhaps consider `await fs.readFile` instead of `fs.readFileSync`?

Reply:
And there's where my JS knowledge stumbles.